06_puppy/patrickmarabeas/mapStore.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+// NewSyncStore returns a pointer to a new instance of the MapStore struct which implements the Storer interface.
+func NewMapStore() Storer {
+	return &MapStore{
+		uuid:  0,
+		store: map[int]Puppy{},
+	}
+}
+
+// Create increments the uuid and adds the provided Puppy struct to the store with this identifier.
+func (store *MapStore) Create(puppy Puppy) int {
+	store.uuid++
+	puppy.ID = store.uuid
+	store.store[store.uuid] = puppy
+
+	return store.uuid
+}
+
+// Read returns the puppy matching the provided uuid.
+// An empty Puppy struct is returned if the identifier does not exist.
+func (store *MapStore) Read(id int) Puppy {
+	if _, ok := store.store[id]; ok {
+		return store.store[id]
+	}
+
+	return Puppy{}
+}
+
+// Update modifies the puppy matching the provided uuid in the store with the provided Puppy struct.
+// No action is performed if the identifier does not exist.
+func (store *MapStore) Update(id int, puppy Puppy) Puppy {
```

Comment:
Returning the input `Puppy` is redundant, which results in extra testing requirements.

Reply:
Simplified to bool.